JumpPads.cs

```diff
@@ -9,33 +9,47 @@ public class JumpPads : UdonSharpBehaviour
 {
     public GameObject referencePoint;
     public float threshold = 5.0f;
-    private float jumpHeight;
-    private bool inProx;
     public float augmentedJumpImpulse = 20;
+    
+    private float jumpHeight;
+    private bool inProx = false;
+    private VRCPlayerApi localPlayer;
+    
+    private void Start()
+    {
+        if (referencePoint == null)
+        {
+            referencePoint = this.gameObject;
+        }
+        localPlayer = Networking.LocalPlayer;
+        jumpHeight = localPlayer.GetJumpImpulse();
+    }
 
     private void Update()
     {
-        var dist = Vector3.Distance(Networking.LocalPlayer.GetPosition(), referencePoint.transform.position);
+        var dist = Vector3.Distance(localPlayer.GetPosition(), referencePoint.transform.position);
         if (dist < threshold)
         {
-            if (!inProx) ProximityAction();
-            inProx = true;
+            if (!inProx)
+            {
+                inProx = true;
+                ProximityAction();
+            }
         }
-        else
+        else if (inProx)
         {
-            if (inProx) ExitProxAction();
             inProx = false;
+            ExitProxAction();
         }
     }
 
-    public void ProximityAction()
+    private void ProximityAction()
     {
-        jumpHeight = Networking.LocalPlayer.GetJumpImpulse();

```

Comment:
Keep this check to revert to jump impulse directly before jumping instead of when script was loaded.

Reply:
Reverted